### ISSUE TITLE
EEx instead of Slime

### ISSUE THREAD

**Issue by tielur** [still-ex/still#124]
How hard would it be to use something like EEx instead of Slime?

**Comment by gabrielpoca** [comment]
Hey @tielur EEx should be working. I have to finish a section on documentation about it, but if you put a file with the extension `.eex` it works like Slime. Or did you try it and it didn't work 🙈?

**Comment by tielur** [comment]
You know... I didn't try 🤦 I just assumed it was hardcoded to Slime because that was the only mix dependency. I'll give EEx a shot today. If I run into anything I'll ask a question here. If you need any help with documentation I can give that a whirl

**Comment by gabrielpoca** [comment]
@tielur it's our fault for not having a single example with eex. I've merged a PR with more docs on that and I'll publish a new version in a bit, feel free to ask questions, it will help us improve the docs!

**Comment by tielur** [comment]
@gabrielpoca what's your thoughts on an argument being passed to `mix still.new` to use EEx when generating the site instead of Slime?

**Comment by tielur** [comment]
`mix still.new my_site --preprocessor eex`
`mix still.new my_site --preprocessor slime`

We could default it to slime to keep backwards compatability. I'm not sure if `preprocessor` is the best argument name

**Comment by tielur** [comment]
If you agree this could be helpful and the argument name can be decided I'm willing to open a PR to add that feature

**Comment by gabrielpoca** [comment]
I agree with you that it's not clear that you can use `.eex` instead of `.slime`, but I'm not sure that adding that option to the generator really achieves that goal. What do you think of a simpler solution, like adding a page built with `.eex` to the generated site? Or simply transforming the `footer.slime` into a `footer.eex`? Would that be enough for the users to understand that you can use both?

**Comment by tielur** [comment]
@gabrielpoca That might be more confusing on why some are slime and some are EEx, I think if the documentation is updated to at least mention that EEx is available out of the box that's probably fine 👍

**Comment by gabrielpoca** [comment]
You're right. I'll make time next week to work on the docs. Just today someone was having issues contributing because they didn't know that you need ImageMagick in your system :s
